templates/repo/diff/box.tmpl

```diff
@@ -60,16 +60,9 @@
 						diffEnd: {{.Diff.End}},
 						link: "{{$.Link}}"
 					};
-					if(window.config.pageData.diffFileInfo) {
-						// Page is already loaded - add the data to our existing data
-						window.config.pageData.diffFileInfo.files.push(...diffData.files);
-						window.config.pageData.diffFileInfo.isIncomplete = diffData.isIncomplete;
-						window.config.pageData.diffFileInfo.diffEnd = diffData.diffEnd;
-						window.config.pageData.diffFileInfo.link = diffData.link;
-					} else {
-						// new load of page - populate initial data
-						window.config.pageData.diffFileInfo = diffData;
-					}
+
+					// this will be read and stored in a reactive store
+					window.config.pageData.diffFileInfo = diffData;
```

Comment:
Here is a longstanding problem (it didn't look good to me from the original PR .... that's why the diff tree broke many times)

* https://github.com/go-gitea/gitea/pull/21012#pullrequestreview-1117876621

The original PR uses `$('body').append($(resp).find('script#diff-data-script'));` to re-execute this script to do some hacky things. I am not sure whether this PR's change still works well with it.

Reply:
I will try to refactor this diff tree "pageData" in a separate PR, to see whether we can get a clear solution.

Reply:
The (nearly) complete fix for this problem could be like this:

-> Refactor diffFileInfo / DiffTreeStore #24998